### PR TITLE
Make input() work in describe.one

### LINKED ISSUE #3650: attribute() doest not work inside describe.one
### 🎛 Description

This is a 🐛 bug report

Pulling attribute values using attribute() does not work inside a describe.one block

### 🌍 InSpec and Platform Version

3.0.61

### 🤔 Replication Case

The following defenition will fail
```
  describe.one do
  	describe http("http://#{attribute('application_url')}", enable_remote_worker: true) do
  	  its('status') { should cmp 200 }
  	end


  	describe http("https://#{attribute('application_url')}", enable_remote_worker: true) do
  	  its('status') { should cmp 200 }
  	end
  end
```
Error:
```
     ×  Control Source Code Error ./controls/example.rb:8
     undefined method `attribute' for #<#<Class:0x00007ff13543abd8>:0x00007ff13543a408>

```

However this works fine:

```
	describe http("http://#{attribute('application_url')}", enable_remote_worker: true) do
	  its('status') { should cmp 200 }
	end
```

### 💻 Stacktrace

```
Traceback (most recent call last):
	23: from /Users/rxavier/.rvm/gems/ruby-2.5.1/bin/ruby_executable_hooks:24:in `<main>'
	22: from /Users/rxavier/.rvm/gems/ruby-2.5.1/bin/ruby_executable_hooks:24:in `eval'
	21: from /Users/rxavier/.rvm/gems/ruby-2.5.1/bin/inspec:23:in `<main>'
	20: from /Users/rxavier/.rvm/gems/ruby-2.5.1/bin/inspec:23:in `load'
	19: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/bin/inspec:12:in `<top (required)>'
	18: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	17: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	16: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	15: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	14: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/cli.rb:245:in `exec'
	13: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/runner.rb:110:in `run'
	12: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/runner.rb:81:in `load'
	11: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/runner.rb:81:in `each'
	10: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/runner.rb:99:in `block in load'
	 9: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile.rb:192:in `collect_tests'
	 8: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile.rb:192:in `each'
	 7: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile.rb:195:in `block in collect_tests'
	 6: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile_context.rb:144:in `load_control_file'
	 5: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile_context.rb:160:in `load_with_context'
	 4: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/profile_context.rb:160:in `instance_eval'
	 3: from ./controls/example.rb:1:in `load_with_context'
	 2: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/describe.rb:19:in `one'
	 1: from /Users/rxavier/.rvm/gems/ruby-2.5.1/gems/inspec-3.0.61/lib/inspec/describe.rb:19:in `instance_eval'
./controls/example.rb:2:in `block in load_with_context': undefined method `attribute' for #<#<Class:0x00007fb219111930>:0x00007fb219111638> (NoMethodError)
```

### LINKED ISSUE #4523: Inputs do not work within describe.one do blocks in version 4.17.7
# Describe the problem
In version 4.17.7 of inspec, inputs do not work with a describe.one block. Example:
```
control 'V-13729' do
  title 'The httpd.conf MaxSpareServers directive must be set properly. '
  apache_conf_file_path = input('apache_conf_file_path', value: '/etc/httpd/conf/httpd.conf', description: 'Location of the configuration file of the Apache service')

  describe.one do
    describe apache_conf(apache_conf_file_path) do
      its('MaxSpareServers') { should cmp <= 10 }
    end
    describe apache_conf(apache_conf_file_path) do
      its('MaxSpareServers') { should cmp == nil }
    end
  end
end
```
results in
```
  ×  V-13729: The httpd.conf MaxSpareServers directive must be set properly.
     ×  Control Source Code Error cerner_Apache_2.2_STIG_inspec/cerner_Apache_2.2_Server_STIG_inspec/controls/V-13729.rb:1
     undefined local variable or method `apache_conf_file_path' for #<#<Class:0x00007f8b02ca7a30>:0x00007f8b01ca4d40>

```

Whereas in version 4.16.0 I can run the same control and it gets this result:
```
 ✔  V-13729: The httpd.conf MaxSpareServers directive must be set properly.
     ✔  Apache Config /etc/httpd/conf/httpd.conf MaxSpareServers should cmp == nil
```

### PR DESCRIPTION
**Pull request by clintoncwolfe**
## Description

Adds the ability for `input()` (and its non-evaluating friend `input_object()`) to work inside a `describe.one` block. 

While this was reported as a regression, I don't believe this functionality was ever in place, either as `input()` or as `attribute()` - reproduction steps with a version welcome on #4523.

Reviewer notes:
This PR currently includes a simple functional test.
~~There is a small amount of code duplication which is noted in a TODO which could be factored out.~~

## Related Issue

Fixes #4523 
Fixes #3650 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by zenspider — APPROVED** [review]
Please file issues for todos

File: lib/inspec/describe_base.rb [hunk 1]
```diff
@@ -17,6 +17,30 @@ def one(&block)
       @action.call("describe.one", @checks, nil)
     end
 
+    def __profile_id
```
Comment (miah) [inline lib/inspec/describe_base.rb]: Is this a private method? Should it be below a `private`?

**Review by miah — CHANGES_REQUESTED** [review]
Looks good just one question about `__profile_id`

File: lib/inspec/describe_base.rb (hunk 1 above)
Comment (clintoncwolfe) [reply lib/inspec/describe_base.rb]: All of these methods end up exposed as part of the DSL inside `describe.one` blocks, which are executed using `instance_eval` - so while we can make this private, it won't actually make any difference. The double-underscore is there to discourage people from accidentally finding it. Same pattern is used in other areas of the DSL code.

I'll add an RDoc comment to that effect.

Comment (miah) [reply lib/inspec/describe_base.rb]: Thanks!

File: lib/inspec/describe_base.rb [hunk 2]
```diff
@@ -17,12 +22,32 @@ def one(&block)
...
     end
 
     def describe(*args, &block)
       @checks.push(["describe", args, block])
     end
+
+    private
+
+    # While this is marked private, it gets consumed during an instance_eval,
+    # so it is fully visible. The double undersroe is there to discourage
```
Comment (miah) [inline lib/inspec/describe_base.rb]: undersCORE. <3

**Review by miah — APPROVED** [review]
Fix typo and this is great. Thanks =)

**Pull request merged**